stage1/init/init.go

```diff
@@ -122,8 +122,18 @@ func getArgsEnv(p *Pod, debug bool) ([]string, []string, error) {
 
 	case "src":
 		args = append(args, filepath.Join(common.Stage1RootfsPath(p.Root), nspawnBin))
-		args = append(args, "--boot") // Launch systemd in the pod
-
+		args = append(args, "--boot")              // Launch systemd in the pod
+		args = append(args, "--register", "false") // We cannot assume the host system is running systemd
+		out, err := os.Getwd()
+		if err != nil {
+			return nil, nil, err
+		}
+		lfd, err := common.GetRktLockFD()
+		if err != nil {
+			return nil, nil, err
+		}
+		args = append(args, fmt.Sprintf("--pid-file=%v", filepath.Join(out, "pid")))
+		args = append(args, fmt.Sprintf("--keep-fd=%v", lfd))
```

Comment:
Since v219, systemd-nspawn has a [patch to do the locks itself](http://cgit.freedesktop.org/systemd/systemd/commit/?id=30535c16924a3da7b47ea87190d929d617d95c5a). Since rkt uses systemd-nspawn v215, we don't use that upstream patch and we use our own patch `stage1/rootfs/usr_from_src/patches/0002-nspawn-add-keep-fd-support.patch` instead.

This would need some research to check if the systemd-nspawn locks and the rkt locks have the same semantic or if rkt needs for locks are covered by the systemd-nspawn locks.

If feasible, I would like to replace the `--keep-fd` by the systemd-nspawn locks upstream. This could be done later, and for now we could just add a comment such as:
```
// TODO: Check if we can replace --keep-fd by systemd-nspawn locks
// See http://cgit.freedesktop.org/systemd/systemd/commit/?id=30535c16924a3da7b47ea87190d929d617d95c5a
```